src/server/middleware/dev.middleware.ts

```diff
@@ -0,0 +1,71 @@
+import { resolve } from 'path';
+import webpack from 'webpack';
+import webpackDevServer from 'webpack-dev-middleware';
+import webpackHotServer from 'webpack-hot-middleware';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+
+const publicPath = '/';
+
+/**
+ * Configures react development for the client.
+ */
+
+const config = {
+  name: 'client',
+  mode: 'development',
+  entry: [
+    'webpack-hot-middleware/client',
+    'react-hot-loader/patch',
+    './src/client/index.js',
+  ],
+  output: {
+    path: resolve(__dirname, 'build/static'),
+    filename: 'app.js',
+    publicPath,
+  },
+  target: 'web',
+  module: {
+    rules: [
+      {
+        test: /\.[jt]sx?$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              plugins: ['react-hot-loader/babel'],
+            },
+          },
+          'ts-loader',
+        ],
+      },
+    ],
+  },
+  resolve: {
+    alias: {
+      'react-dom': '@hot-loader/react-dom',
+    },
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      title: process.env.APP_NAME,
+      template: resolve(__dirname, '.src/client/index.html'),
```

Comment:
To resolve the path issues you were having here - what do you think of exporting `SRC` from webpackfile.js? And then importing it here?

Reply:
e355ee0 adds another webpack plugin that injects the `<div id="root">` so we no longer need the html template.